velox/buffer/Buffer.h

```diff
@@ -116,7 +116,9 @@ class Buffer {
   // TODO: `resize` is probably a better name for this method
   virtual void setSize(size_t size) {
     VELOX_CHECK(mutable_);
-    VELOX_CHECK_LE(size, capacity_);
```

Comment:
The function looks like aims to set to a little size, revoke function should guarantee not exceed capacity, is this flow reasonable?

Reply:
This was only a quick fix. After some investigation, I made a new fix in this PR. Maybe you could take a review, thanks.